### PR TITLE
Add Q16.16 fixed-point representation

### PR DESCRIPTION
**Pull request by bessman**
This PR adds a simple implementation of Q16.16 fixed-point arithmetic, intended for use by the digital multimeter instrument to abstract away ADC details like calibration offset and reference voltage.

## Summary by Sourcery

Add a Q16.16 fixed-point arithmetic module to the utility library, providing a consistent representation for ADC-related calculations, with conversion utilities, arithmetic functions supporting saturation and rounding, part extraction, and formatted string output, and integrate comprehensive unit tests and build support.

New Features:
- Implement Q16.16 fixed-point arithmetic library with conversion macros, runtime conversion, arithmetic operations (add, sub, mul, div) with saturation and rounding, part extraction, and string formatting

Build:
- Include fixed_point.c in the pslab-util library and add test_fixed_point to the CMake test suite

Tests:
- Add comprehensive unit tests for fixed-point conversions, constants, arithmetic operations, edge cases (overflow, underflow, division by zero), rounding behavior, boundary conditions, math properties, and string output

### CODE REVIEW

File: src/util/fixed_point.h [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+ * @brief Fixed-point division
+ *
+ * Uses 64-bit intermediate to maintain precision.
+ */
+static inline Fixed fixed_div(Fixed a, Fixed b)
+{
+    if (b == 0) {
+        return (a >= 0) ? INT32_MAX : INT32_MIN;
+    }
+    int64_t temp = ((int64_t)a << FIXED_FRAC_BITS) / (int64_t)b;
```
[the remark this answers is not in the document]
Comment (bessman) [reply src/util/fixed_point.h]: > int64_t temp = (((int64_t)a << FIXED_FRAC_BITS) + ((b > 0 ? b : -b) / 2)) / (int64_t)b;

Is this correct?

[the remark this answers is not in the document]
Comment (bessman) [reply src/util/fixed_point.h]: It's not correct for negative numerators. By always adding abs(b), the result is biased toward zero for negative numerators. For example, with a = -5 and b = 3, the proposed algorithm yields -1.666656 instead of the expected -1.666672.

**Comment by bessman** [comment]
@sourcery-ai review

File: src/util/fixed_point.c [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+char *FIXED_to_string(
+    FIXED_Q1616 const x,
+    char *const buffer,
+    size_t const buffer_size
+)
+{
+    unsigned const buffer_min_size = 13;
+    if (buffer == nullptr || buffer_size < buffer_min_size) {
+        return nullptr;
+    }
```
[the remark this answers is not in the document]
Comment (bessman) [reply src/util/fixed_point.c:26]: 13 is sufficient for all values. Worst case is 1 minus sign, 5 integer digits, 1 decimal point, 5 decimal places, and 1 null terminator. 1 + 5 + 1 + 5 + 1 == 13.

File: tests/test_fixed_point.c [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+    TEST_ASSERT_EQUAL(INT32_MIN, FIXED_sub(INT32_MIN, 1));
+
+    // Test float conversion boundaries
+    float max_float = FIXED_TO_FLOAT(INT32_MAX);
+    float min_float = FIXED_TO_FLOAT(INT32_MIN);
+
+    TEST_ASSERT_TRUE(max_float > 30000.0f); // Much larger than 32767
+    TEST_ASSERT_TRUE(min_float < -30000.0f); // Much smaller than -32768
+}
+
```
[the remark this answers is not in the document]
Comment (bessman) [reply tests/test_fixed_point.c:322]: FIXED_FROM_FLOAT is documented to not support out of range input.
